docs/resources/server/loottables/datagen.md

```diff
@@ -0,0 +1,200 @@
+        super(output,
+                // A set of required table resource locations. This is used in validation (see below).
+                // It is generally not recommended for mods to validate, therefore we pass in an empty set.
+                Set.of(),
+                // A list of sub provider entries. See below for what values to use here.
+                List.of());
+    }
+    
+    // Vanilla uses this method to validate existence and correct usage of loot context parameters.
+    // This is a pretty complex setup, so it is usually recommended for modders to trust themselves and no-op here.
+    @Override
+    protected void validate(Map<ResourceLocation, LootTable> map, ValidationContext context) {}
+}
+```
+
+Like all data providers, we register the provider to `GatherDataEvent`:
+
+```java
+@SubscribeEvent
+public static void onGatherData(GatherDataEvent event) {
+    event.getGenerator().addProvider(event.includeServer(), MyLootTableProvider::new);
+}
+```
+
+## `LootTableSubProvider`s
+
+`LootTableSubProvider`s are where the actual generation happens. To get started, we implement `LootTableSubProvider` and override `#generate`:
+
+```java
+public class MyLootTableSubProvider implements LootTableSubProvider {
+    @Override
+    public void generate(BiConsumer<ResourceLocation, LootTable.Builder> consumer) {
+        // LootTable.lootTable() returns a loot table builder we can add loot tables to.
+        consumer.accept(new ResourceLocation(ExampleMod.MOD_ID, "example_loot_table"), LootTable.lootTable()
+                // Add a loot table-level loot function. This example uses a number provider (see below).
+                .apply(SetItemCountFunction.setCount(ConstantValue.exactly(5)))
+                // Add a loot pool.
+                .withPool(LootPool.lootPool()
+                        // Add a loot pool-level function, similar to above.
+                        .apply(...)
+                        // Add a loot pool-level condition. This example only rolls the pool if it is raining.
+                        .when(WeatherCheck.weather().setRaining(true))
+                        // Set the amount of rolls and bonus rolls, respectively.
+                        // Both of these methods utilize a number provider.
+                        .setRolls(UniformGenerator.between(5, 9))
+                        .setBonusRolls(ConstantValue.exactly(1))
+                        // Add a loot entry. This example returns an item loot entry. See below for more loot entries.
+                        .add(LootItem.lootTableItem(Items.DIRT))
+                )
+        );
+    }
+}
+```
+
+Once we have our loot table sub provider, we add it to the constructor of our loot provider, like so:
+
+```java
+super(output, Set.of(), List.of(new SubProviderEntry(
+        // A reference to the sub provider's constructor.
+        MyLootTableSubProvider::new,
+        // An associated loot context set. If you're unsure what to use, use empty.
```

Comment:
This is terrible advice. Actually explain what the loot context param set represents and is for.